targets/esp8266/Makefile.travis

```diff
@@ -0,0 +1,51 @@
+# Copyright JS Foundation and other contributors, http://js.foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Default target for running the build test outside the Travis CI environment.
+all:
+	$(MAKE) install
+	$(MAKE) script
+
+
+## Targets for installing build dependencies of the ESP8266 JerryScript target.
+
+# Install tools via apt.
+install-apt-get-deps:
+	sudo apt-get install -q -y gperf texinfo wget
+
+# Fetch and build crosstool-NG with support for Xtensa.
+install-xtensa-kx106-gcc:
+	git clone https://github.com/jcmvbkbc/crosstool-NG.git ../crosstool-NG -b lx106-g++-1.21.0
+	cd ../crosstool-NG && ./bootstrap
+	cd ../crosstool-NG && ./configure --enable-local
+	$(MAKE) -C ../crosstool-NG --no-print-directory -Rr # HACK: without the command line arguments, make bails out with recursion error
+	cd ../crosstool-NG && ./ct-ng xtensa-lx106-elf
+	cd ../crosstool-NG && ./ct-ng build
+
+# Fetch Espressif SDK and Xtensa libraries.
+install-espressif-sdk:
+	git clone https://github.com/espressif/ESP8266_RTOS_SDK.git ../ESP8266_SDK
+	cd ../ESP8266_SDK && git checkout 2fab9e23d779cdd6e5900b8ba2b588e30d9b08c4
+	wget https://github.com/esp8266/esp8266-wiki/raw/master/libs/libhal.a -O ../ESP8266_SDK/lib/libhal.a
+
+# Perform all the necessary (JerryScript-independent) installation steps.
+install: install-apt-get-deps install-xtensa-kx106-gcc install-espressif-sdk
+
+
+## Targets for building ESP8266 with JerryScript.
+
+# Build the firmware (ESP8266 with JerryScript).
+script:
```

Comment:
I'd add `install` target as a dependency of this target to prevent the parallel "build" of `script` and `install`

Reply:
If you agree in my opinion, then please do it in the other new makefiles too.

Reply:
That never happens (I mean the parallel execution of script and install). If `all` is called as the default target of make, then it explicitly lists the order of execution by invoking sub-makes in a specific order (install, script). While `.travis.yml` calls make twice, first with `install` target in the `install` stage and then with `script` target in script stage. (The make targets intentionally reflect the names of the Travis CI job stage names to highlight which target is called in which stage.) If `install` was a dependency of `script`, `install` would run again in script stage.

See:
```yml
    - env: JOBNAME="ESP8266 Build Test"
      cache: ccache
      install: make -f ./targets/esp8266/Makefile.travis install
      script: make -f ./targets/esp8266/Makefile.travis script
```

Reply:
I see now, why you did it in this way. It is good, leave as is.